compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -1157,6 +1157,27 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    #[instrument(skip(self, err))]
+    pub(crate) fn suggest_floating_point_literal(
+        &self,
+        err: &mut Diagnostic,
+        expr: &hir::Expr<'_>,
+        expected_ty: Ty<'tcx>,
+    ) -> bool {
+        if let ExprKind::Struct(QPath::LangItem(LangItem::Range, ..), [start, _], _) = expr.kind
+            && expected_ty.is_floating_point()
+        {
+            err.span_suggestion_verbose(
+                self.tcx.sess.source_map().next_point(start.span),
```

Comment:
Instead of this, can we do `start.span.until(end.span)` and change the suggestion snippet to `.`? This will be slightly more resilient in macros and much more resilient to weird formatting (like `start   ..   end`).